Add fallback service credentials for current user authentication (#1792)
grafana/azure-data-explorer-datasource: doc/current-user-auth.md

user_identity_client_secret = 4479f5a6-444c-4271-8790-60eeb42225ae
```

> ## What doesn't work (and how to fix it)

Reviewer: ```suggestion
## Current user authentication and background features
```

> With current user authentication, every query runs as the Grafana user who is signed in. That's fine when someone is looking at a dashboard, but some Grafana features run in the background with nobody signed in. The main ones are **alerting**, **recorded queries**, and **reporting**. These have no user to run as, so they fail.

Reviewer: ```suggestion

With current user authentication, every query runs as the Grafana user who is signed in. That works when someone is viewing a dashboard, but some Grafana features run in the background with no signed-in user. The main ones are **alerting**, **recorded queries**, and **reporting**. Because these features have no user to run as, they fail.

```

> To keep them working, you can set up **fallback service credentials**.

Reviewer: ```suggestion
To keep them working, set up **fallback service credentials**.
```

## Fallback service credentials

> Fallback credentials are a shared identity (an App Registration, managed identity, or workload identity) that the data source falls back to whenever a request has no signed-in user behind it. Turn them on and your background features (like alerting) keep working instead of failing.

Reviewer: ```suggestion
Fallback credentials are a shared identity (an App Registration, managed identity, or workload identity) that the data source uses whenever a request has no signed-in user. When you enable them, background features such as alerting continue to work instead of failing. 
```

> > **Note:** Anything that uses the fallback runs as this shared identity, not as an individual user. So it can see different data than a person querying interactively would.

Reviewer: ```suggestion
{{< admonition type="note" >}}
Requests that use the fallback run as this shared identity rather than as an individual user. As a result, they might return different data than an interactive query run by a specific user.
{{< /admonition >}}
```

> Setting this up takes three steps.

Reviewer: ```suggestion
To set up fallback credentials, complete the following three steps.
```

> ### 1. Turn fallback on in the Grafana config

Reviewer: ```suggestion
### 1. Enable fallback credentials in the Grafana configuration
```

> In the `[azure]` section, enable fallback credentials. (This is already `true` by default when `user_identity_enabled` is on, but it's fine to set it explicitly.)

Reviewer: ```suggestion
In the `[azure]` section of your Grafana configuration file, enable fallback credentials. This setting is `true` by default when `user_identity_enabled` is set, but you can also set it explicitly.
```

```ini
[azure]
user_identity_enabled = true
user_identity_fallback_credentials_enabled = true
```

> ### 2. Turn on ID forwarding

Reviewer: ```suggestion
### 2. Enable ID forwarding
```

> Grafana needs a way to tell a real user's request apart from a background job. The `idForwarding` feature toggle gives it that signal:

Reviewer: ```suggestion
Grafana needs a way to distinguish a signed-in user's request from a background job. The `idForwarding` feature toggle provides that signal.
```

```ini
[feature_toggles]
idForwarding = true
```

Without `idForwarding`, Grafana can't recognize background requests (like alert evaluations) as having no user, so it never switches to the fallback, and those requests fail.

> ### 3. Add the credentials in the data source

Reviewer: ```suggestion
### 3. Add the credentials to the data source
```

> In the data source settings, choose **Current User**, switch **Fallback service credentials** to **Enabled**, pick the identity type (App Registration, Managed Identity, or Workload Identity), and fill in the details.

Reviewer: ```suggestion
In the data source settings:

1. Set **Authentication** to **Current User**.
1. Set **Fallback service credentials** to **Enabled**.
1. Select the identity type: **App Registration**, **Managed Identity**, or **Workload Identity**.
1. Enter the credential details for the identity type you selected.
```
